irohad/torii/impl/command_service_impl.cpp

```diff
@@ -150,6 +152,27 @@ namespace iroha {
 
     void CommandServiceImpl::processBatch(
         std::shared_ptr<shared_model::interface::TransactionBatch> batch) {
+      auto cache_presence = tx_presence_cache_->check(*batch);
+      if (not cache_presence) {
+        // TODO andrei 30.11.18 IR-51 Handle database error
+        log_->warn("Check tx presence database error. {}", *batch);
+        return;
+      }
+      auto is_replay = std::any_of(
```

Comment:
It would be nice to place a TODO to make a method to `TxPresenceCache` that generalizes this check, because there are at least 5 analogous places in the project.

Reply:
Caches lookup order was changed and the code was modified. 
Now it could not be generalized with the rest part of replay checks.